eos-phone-home

```diff
@@ -0,0 +1,398 @@
+
+        return product
+
+    def _get_image(self):
+        image = None
+
+        for path in ['/sysroot', '/']:
+            try:
+                image = subprocess.check_output(
+                            ['attr', '-q', '-g', 'eos-image-version', path])
+                image = image.decode(errors='replace').strip()
+
+                print(" - Image from %s: %s" % (path, image))
+                return image
+            except Exception as e:
+                print("Unable to get %s image name!" % (path, ), e,
+                      file=sys.stderr)
+
+        return None
+
+    def _get_release(self):
+        release = None
+
+        try:
+            release_data = None
+            with open('/etc/os-release') as release_file:
+                release_data = release_file.read()
+
+            for release_item in release_data.splitlines():
+                release_item = release_item.strip()
+
+                key, value = release_item.split('=', 1)
+                if key == 'VERSION':
+                    print(' - Found version key')
+                    if value.startswith('"') and value.endswith('"'):
+                            value = value[1:-1]
+
+                    release = value
+                    print(' - Release:', release)
+                    return release
+
+        except Exception as e:
+            print("Unable to get release name!", e, file=sys.stderr)
+
+        if release:
+            release = release.strip()
+
+        return release
+
+    def _get_serial(self):
+        serial = None
+
+        try:
+            if os.path.isdir('/sys/class/dmi/id'):
+                with open('/sys/class/dmi/id/product_serial') as ssn_file:
+                    serial = ssn_file.read().strip()
+            else:
+                with open('/sys/class/endless_mfgdata/entries/SSN') \
+                  as ssn_file:
+                    serial = ssn_file.read().strip()
+        except Exception as e:
+            print("Unable to get serial! Check that you're running as root!",
+                  e, file=sys.stderr)
+
+        print(" - Found serial:", serial)
+
+        if serial:
+            serial = serial.strip()
+
+        return serial
+
+    def _get_live(self):
+        live = False
+
+        command_line = None
+        with open('/proc/cmdline', 'r') as command_line_file:
+            command_line = command_line_file.read()
+
+        if command_line:
+            command_line = command_line.strip()
+
+        print(" - Cmdline:", command_line)
+
+        if re.search(r'\bendless.live_boot\b', command_line):
+            print(" - Found live marker. Marking as live image.")
+            live = True
+
+        return live
+
+    def _get_count(self):
+        count = None
+
+        try:
+            with open(os.path.join(self.STATE_DIRECTORY, 'count'), 'r') \
+              as count_file:
+                count_data = count_file.read()
+
+                if count_data:
```

Comment:
This and following lines should ideally be outside of `with` block so that we don't keep the file open for longer than necessary.